src/net_processing.cpp

```diff
@@ -1982,7 +1982,7 @@ static void ProcessHeadersMessage(CNode& pfrom, CConnman& connman, ChainstateMan
             }
         }
 
-        if (!pfrom.fDisconnect && pfrom.IsOutboundOrBlockRelayConn() && nodestate->pindexBestKnownBlock != nullptr && pfrom.m_tx_relay != nullptr) {
+        if (!pfrom.fDisconnect && pfrom.IsFullOutboundConn() && nodestate->pindexBestKnownBlock != nullptr) {
```

Comment:
ok, I've looked into this and am happy to discuss with you, but want to clarify we are on the same page that **this PR doesn't change the behavior.**

I agree that the comment initially seems inconsistent with the logic in `ConsiderEviction`, but I'm still trying to wrap my head around all the mechanisms (since the chainwork check is repeated as the first conditional in `ConsiderEviction`, is the protection here essential?). I have many observations & questions, but I don't think this PR is the right place to have this conversation. are you willing to open an issue?

Reply:
Yes as far as I understand this code it is neither introduced by this PR or #19316. 

>  (since the chainwork check is repeated as the first conditional in ConsiderEviction, is the protection here essential?).

You mean the check `pindexBestKnownBlock != nullptr` ? But you have a conditional above and it won't exclude block-relay-only with `m_protect=false`. I'm not sure about the exact semantic of `m_protect` but yes I should be able to write a test for this with #19315 to assert bug :)

Reply:
Looks like a documentation oversight when I changed around how this code worked while working on #15759. See https://github.com/bitcoin/bitcoin/pull/15759#pullrequestreview-281092623.  I think originally I was going to make block-relay-only peers immune from all eviction, and then changed it to only be immune to outbound peer rotation-based eviction.

Reply:
Tracked in https://github.com/bitcoin/bitcoin/issues/19863#issue-692406060, either to document block-relay-only eviction expectations or align protection scope among all our outbound peers.